source/CommonJobs/CommonJobs.MVC.UI/Scripts/Evaluations/admin.js

```diff
@@ -89,29 +97,47 @@
     $('#generate-evaluation-button').on('click', function () {
         var model = viewmodel.toJs();
         var modelFiltered = { Employees: _.filter(model.Employees, function (e) { return e.ResponsibleId && !e.Period; }) };
-        $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
-            type: "POST",
-            dataType: 'json',
-            contentType: 'application/json; charset=utf-8',
-            data: JSON.stringify(modelFiltered),
-            complete: function (response) {
-                var modalContainer = $('#evaluations-generated-confirm');
-                var countText = (response.responseText == '1') ? "Se ha generado 1 evaluación correctamente" : "Se han generado " + response.responseText + " evaluaciones correctamente";
-                modalContainer.find('#textCount').text(countText);
-                modalContainer.modal('show');
-            },
-            error: function () {
-                alert('Fallo interno. Por favor recargue la página.');
-            }
-        });
+        if (!viewmodel.isInvalid()) {
+            $.ajax("/Evaluations/api/GenerateEvalutions/" + evaluationPeriod + "/", {
+                type: "POST",
+                dataType: 'json',
+                contentType: 'application/json; charset=utf-8',
+                data: JSON.stringify(modelFiltered),
+                complete: function (response) {
+                    var modalContainer = $('#evaluations-generated-confirm');
+                    var title = "EVALUACIONES GENERADAS"
+                    modalContainer.find('#title').text(title);
```

Comment:
If the modal is supposed to be a component to reuse (and this seems to be "reusing it", which is great), it should not have ids, but only classes internally. Components should be prepared to appear more than once in the same page, if necessary.